salt/states/file.py

```diff
@@ -3519,7 +3519,7 @@ def append(name,
 
     with salt.utils.fopen(name, 'rb') as fp_:
         slines = fp_.readlines()
-        slines = [item.rstrip() for item in slines]
+        slines = [item.rstrip(os.linesep) for item in slines]
```

Comment:
If you want, maybe consider replacing these two lines with just `slines = fp_.read().splitlines()`. It's cleaner and [`splitlines()`](https://docs.python.org/2/library/stdtypes.html#str.splitlines) natively handles all line separators. Do the same for `nlines` and it should be good.